### ISSUE TITLE
Should the registration of OCC commands for app developers be documented?

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/documentation#5460]
I am part of the cookbook dev team and we were discussion adding a command to the `occ` cli. In the official documentation there is no mention of it thus I thought it not to be possible.  Another dev corrected me.

The question is now simply: Is it intentional to avoid extensive usage of the CLI for whatever reason or just nobody thought of it or had the time?

I want to ask before writing some documentation and opening a PR just to see it rejected.

**Comment by MorrisJobke** [comment]
> The question is now simply: Is it intentional to avoid extensive usage of the CLI for whatever reason or just nobody thought of it or had the time?

Not really. It's totally fine to create commands. Just keep in mind that there are setups where OCC is not really available (shared hosters for example), but beside that the usage is totally fine :(

**Comment by skjnldsv** [comment]
Confirmed gap. The `developer_manual/app_development/` section has no documentation on adding occ commands to apps. `info.rst` covers the app metadata registration (`commands/command` pointing to a PHP class) but there is nothing explaining how to implement the class, what interface to use (`OCP\Command\ICommand` or Symfony Console's `Command`), or how to wire it up.

This is a common app development need and should be a dedicated page or section in the developer manual.

**Comment by skjnldsv** [comment]
Misunderstood, ICommand is deprecated and reserved for internal NC usage
